### ISSUE TITLE
"Adaptive" ephemeris markers

### ISSUE THREAD

**Issue by Atque** [Stellarium/stellarium#2829]
I'm not sure about the terminology here, but I sometimes find the ephemeris markers too dense AND too sparse at the same time when showing an ephemeris, because they are currently only showing a fixed interval. What I'm wishing for, is for some kind of "adaptive" markers. See the screenshot below, of comet C/1990 Levy.

![stellarium-149](https://user-images.githubusercontent.com/37028368/200116156-d4becc0e-137a-4278-a040-536475d2ec03.png)

You can see how the markers are too dense near the standstill/retrograde, and too sparse when the comet "gains speed" later in the year. The current solution makes it hard to create and visualize ephemerides for certain bodies.

The website [in-the-sky.org](https://in-the-sky.org/findercharts3.php?id=P5) has ephemeris plotting that really shows what I wish for, see the screenshot below for Mercury:

![mercury_ephem](https://user-images.githubusercontent.com/37028368/200116256-d313065a-316d-4858-a32a-01aa592f1f86.PNG)
Note how the labels show different time intervals, as not to collide with other labels.

It would also be suitable if the markers were adaptive to zooming as well. I.e more density when zooming in.

**Comment by gzotti** [comment]
I only know (printed) ephemerides with fixed steps. This immediately shows where the object runs slower or faster. Adapting label density could be useful.

**Comment by Atque** [comment]
Yes, label density is what I mean, not ephemeris steps. So you can calculate eg Mercurys positions from 2022-01-01 to 2022-12-31 with one-day steps, but showing labels as not to make them collide with one another.
